engine/pkg/client/executor_client.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/engine/enginepb"
+	"github.com/pingcap/tiflow/pkg/security"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+// ExecutorClient is the public interface for an executor client.
+// It is only for one executor. For a client to a group of executors,
+// refer to ExecutorGroup.
+type ExecutorClient interface {
+	// An executor's gRPC server exposes two services for now:
+	// ExecutorService and BrokerService. This design is for better decoupling
+	// and for mocking these two services separately.
+
+	ExecutorServiceClient
+	BrokerServiceClient
+
+	// Close closes the gRPC connection used to create the client.
+	Close()
+}
+
+type executorClientImpl struct {
+	ExecutorServiceClient
+	BrokerServiceClient
+
+	conn *grpc.ClientConn
+}
+
+// NewExecutorClient creates a new executor client.
+// Note that conn will be closed if the returned client is closed.
+func NewExecutorClient(conn *grpc.ClientConn) ExecutorClient {
+	return &executorClientImpl{
+		ExecutorServiceClient: NewExecutorServiceClient(enginepb.NewExecutorClient(conn)),
+		BrokerServiceClient:   NewBrokerServiceClient(enginepb.NewBrokerServiceClient(conn)),
+		conn:                  conn,
+	}
+}
+
+// Close closes the gRPC connection maintained by the client.
+func (c *executorClientImpl) Close() {
```

Comment:
Why not propagate the error to its caller?

Reply:
- There is probably no easy remedy for any error that could be returned here.
- If we have an alert system, we can report the error so that we can discover a problem if errors occur frequently.